iModelCore/GeomLibs/serialization/src/FlatBuffer/FixedStructs.cpp

```diff
@@ -1621,6 +1658,8 @@ static ICurvePrimitivePtr ReadCurvePrimitive (const BGFB::VariantGeometry * fbGe
             {
             BGFB::TransitionSpiral const *fbTransitionSpiral
                 = reinterpret_cast <const BGFB::TransitionSpiral *> (fbGeometry->geometry ());
+            if (!fbTransitionSpiral)
+                return nullptr;

```

Comment:
This change is fine. But please also add an early out after the next line when `detail` is nullptr.